src/components/Table/components/HeaderCell/CellContent.jsx

```diff
@@ -7,16 +7,23 @@ import HeaderCellMenu from "./HeaderCellMenu";
 const CellContent = ({
   children,
   isSortable,
+  isAddEnabled,
   sortedInfo,
   onSort,
   isHidable = true,
   onColumnHide,
+  onAddColumn,
+  onColumnDelete,
   column,
   ...headerProps
 }) => {
   const isColumnHidable = isHidable && isPresent(onColumnHide);
+  const isColumnDeletable = isPresent(column?.isSystem) && !column?.isSystem;
```

Comment:
Instead of `isSystem`, can we use `isDeletable`? I think it's more general and can work everywhere. `isSystem` is okay for custom fields, but sometimes it might not make sense. If we use `isDeletable`, it will match with `isHideable` and be consistent. Let's set `isDeletable` to true by default. Also, only enable `isDelete` if `onColumnDelete` is defined. In short, it should work just like `isHideable`.

Or, has the use of the `isSystem` approach been decided based on some discussion? I don't see any discussion in the issue.

Reply:
Definitely agree with you on that. The variable was decided to be `isSystem` simply because that is the convention followed by `neeto-fields-nano` to differentiate between custom fields and system fields. I can update the name to `isDeletable` here but we will have to do more work on the host application to make it into this form.

Reply:
@akkshaylawrence As discussed, please check the amount of work we need in the host application.

Reply:
@josephmathew900 The changes needed are trivial for the products I've tested on. I have updated the code to use the `isDeletable` flag.